src/lib/stream/chacha/chacha.cpp

```diff
@@ -173,4 +172,27 @@ std::string ChaCha::name() const
    return "ChaCha(" + std::to_string(m_rounds) + ")";
    }
 
+void ChaCha::seek(u64bit offset)
+   {
+   if (m_state.size() == 0 && m_buffer.size() == 0)
+      {
+         throw Invalid_State("You have to setup the stream cipher (key and iv)");
+      }
+
+   m_position = offset;
```

Comment:
m_position is the read index into the buffer - should `offset` be reduced modulo the block size here?